Header.hpp

```diff
@@ -3,30 +3,294 @@
+    Q_PROPERTY(int enable_detection MEMBER enable_detection)
+    Q_PROPERTY(QString det_config MEMBER det_config)
+    Q_PROPERTY(QString det_checkpoint MEMBER det_checkpoint)
+    Q_PROPERTY(int enable_human_pose MEMBER enable_human_pose)
+    Q_PROPERTY(int enable_animal_pose MEMBER enable_animal_pose)
+    Q_PROPERTY(int enable_hand_pose MEMBER enable_hand_pose)
+    Q_PROPERTY(QString human_pose_config MEMBER human_pose_config)
+    Q_PROPERTY(QString human_pose_checkpoint MEMBER human_pose_checkpoint)
+    Q_PROPERTY(QVector<int> human_det_ids MEMBER human_det_ids)
+    Q_PROPERTY(QString animal_pose_config MEMBER animal_pose_config)
+    Q_PROPERTY(QString animal_pose_checkpoint MEMBER animal_pose_checkpoint)
+    Q_PROPERTY(QVector<int> animal_det_ids MEMBER animal_det_ids)
+    Q_PROPERTY(QString hand_pose_config MEMBER hand_pose_config)
+    Q_PROPERTY(QString hand_pose_checkpoint MEMBER hand_pose_checkpoint)
+    Q_PROPERTY(QVector<int> hand_det_ids MEMBER hand_det_ids)
+    Q_PROPERTY(QString device MEMBER device)
+    Q_PROPERTY(double det_score_thr MEMBER det_score_thr)
+    Q_PROPERTY(double kpt_thr MEMBER kpt_thr)
+    Q_PROPERTY(int vis_mode MEMBER vis_mode)
+    Q_PROPERTY(bool sunglasses MEMBER sunglasses)
+    Q_PROPERTY(bool bugeye MEMBER bugeye)
+    Q_PROPERTY(QVariant out_video_file MEMBER out_video_file)
+    Q_PROPERTY(int out_video_fps MEMBER out_video_fps)
+    Q_PROPERTY(int buffer_size MEMBER buffer_size)
+    Q_PROPERTY(int inference_fps MEMBER inference_fps)
+    Q_PROPERTY(int display_delay MEMBER display_delay)
+    Q_PROPERTY(bool synchronous_mode MEMBER synchronous_mode)
+
+public:
+    QString name = "Hell8o"; // Default value for the "foo" property
+    int params = 123; // Default value for the "bar" property
+
+    QString cam_id;
+    int enable_detection = 0;
+    QString det_config;
+    QString det_checkpoint;
+    int enable_human_pose = 0;
+    int enable_animal_pose = 0;
+    int enable_hand_pose = 0;
+    QString human_pose_config;
+    QString human_pose_checkpoint;
+    QVector<int> human_det_ids;
+    QString animal_pose_config;
+    QString animal_pose_checkpoint;
+    QVector<int> animal_det_ids;
+    QString hand_pose_config;
+    QString hand_pose_checkpoint;
+    QVector<int> hand_det_ids;
+    QString device;
+    double det_score_thr = 0.0;
+    double kpt_thr = 0.0;
+    int vis_mode = 0;
+    bool sunglasses = false;
+    bool bugeye = false;
+    QVariant out_video_file;
+    int out_video_fps = 0;
+    int buffer_size = 0;
+    int inference_fps = 0;
+    int display_delay = 0;
+    bool synchronous_mode = false;
+
+
+    bool operator==(const Mmpose_params& other) const noexcept = default;
+
+    bool operator<(const Mmpose_params& other) {
+        // Implement less-than comparison
+        // For example, compare just the 'name' and 'params' members
+        return std::tie(name, params) < std::tie(other.name, other.params);
+    }
+};
+
+struct Size {
```

Comment:
I wonder if we couldn't use QSize / QSizeF for this

Reply:
oh ! ok replaced